scripts/sanity_chk/sanitylib.py

```diff
@@ -3009,88 +3189,52 @@ def add_tasks_to_queue(self, build_only=False, test_only=False):
                     instance.status = None
                     pipeline.put({"op": "cmake", "test": instance})
 
-        return "DONE FEEDING"
-
-    def execute(self):
-
-        def calc_one_elf_size(instance):
-            if instance.status not in ["error", "failed", "skipped"]:
-                if instance.platform.type != "native":
-                    size_calc = instance.calculate_sizes()
-                    instance.metrics["ram_size"] = size_calc.get_ram_size()
-                    instance.metrics["rom_size"] = size_calc.get_rom_size()
-                    instance.metrics["unrecognized"] = size_calc.unrecognized_sections()
-                else:
-                    instance.metrics["ram_size"] = 0
-                    instance.metrics["rom_size"] = 0
-                    instance.metrics["unrecognized"] = []
-
-                instance.metrics["handler_time"] = instance.handler.duration if instance.handler else 0
-
+    def pp(self, pipeline, done, lock, results):
+        while True:
+            try:
+                task = pipeline.get_nowait()
+            except queue.Empty:
+                break
+            else:
+                test = task['test']
+                pb = ProjectBuilder(self,
+                                    test,
+                                    lsan=self.enable_lsan,
+                                    asan=self.enable_asan,
+                                    ubsan=self.enable_ubsan,
+                                    coverage=self.enable_coverage,
+                                    extra_args=self.extra_args,
+                                    device_testing=self.device_testing,
+                                    cmake_only=self.cmake_only,
+                                    cleanup=self.cleanup,
+                                    valgrind=self.enable_valgrind,
+                                    inline_logs=self.inline_logs,
+                                    generator=self.generator,
+                                    generator_cmd=self.generator_cmd,
+                                    verbose=self.verbose,
+                                    warnings_as_errors=self.warnings_as_errors
+                                    )
+                pb.process(pipeline, done, task, lock, results)
+
+        return True
+
+    def execute(self, pipeline, done, results):
+        lock = Lock()
         logger.info("Adding tasks to the queue...")
-        # We can use a with statement to ensure threads are cleaned up promptly
-        with BoundedExecutor(bound=self.jobs, max_workers=self.jobs) as executor:
-
```

Comment:
The removed code below has comments explaining what was going on, and it seems the new code lacks commenting.  I would really appreciate it if you would add some comments in the code explaining what is going on there.